### PR TITLE
wiki-tui: fix compilation with rust 1.80

### PR DESCRIPTION
**Pull request by Builditluc**
## Description of changes

Bumps the `time` dependency to allow compilation with rust 1.80 (https://github.com/NixOS/nixpkgs/issues/332957)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by alyssais** [comment]
This seems to be updating lots of dependencies, not just time. (`cargo update time`)  Is there a reason for that?

File: pkgs/misc/wiki-tui/default.nix [hunk 1]
```diff
@@ -19,25 +20,29 @@ rustPlatform.buildRustPackage rec {
...
   buildInputs = [
     ncurses
     openssl
-  ] ++ lib.optionals stdenv.isDarwin [
-    Security
-  ];
+  ] ++ lib.optionals stdenv.isDarwin [ Security ];
 
-  cargoHash = "sha256-rKTR7vKt8woWAn7XgNYFiWu4KSiZYhaH+PLEIOfbNIY=";
+  # Note: bump `time` dependency to be able to build with rust 1.80, should be removed on the next
```
Comment (SuperSandro2000) [inline pkgs/misc/wiki-tui/default.nix:30]: Is this already done upstream or a Pr open for that? If not would you be so kind.

**Comment by Builditluc** [comment]
Don't know, I've updated time with `cargo update time`. But will look into it

File: pkgs/misc/wiki-tui/default.nix [hunk 2]
```diff
@@ -19,25 +20,29 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-euyg4wYWYerYT3hKdOCjokx8lJldGN7E3PHimDgQy3U=";
   };
```
Comment (SuperSandro2000) [inline pkgs/misc/wiki-tui/default.nix:22]: Normally patches and cargoPatches would be placed close to src.

File: pkgs/misc/wiki-tui/default.nix [hunk 3]
```diff
@@ -19,25 +20,29 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-euyg4wYWYerYT3hKdOCjokx8lJldGN7E3PHimDgQy3U=";
```
Comment (SuperSandro2000) [inline pkgs/misc/wiki-tui/default.nix:20]: Also we no longer do repo = pname 😅

**Review by SuperSandro2000 — COMMENTED** [review]
Formatting changes should be done in an extra commit.

File: pkgs/misc/wiki-tui/default.nix (hunk 1 above)
Comment (Builditluc) [reply pkgs/misc/wiki-tui/default.nix:30]: (I'm the maintainer of the package) Upstream has a lot of changes already in `main`, most of which aren't compatible with 0.8.2 (changing the UI framework)

**Comment by alyssais** [comment]
> Don't know, I've updated time with `cargo update time`. But will look into it

It probably doesn't matter that much — don't waste too much time on it.

File: pkgs/misc/wiki-tui/default.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/misc/wiki-tui/default.nix:30]: > Is this already done upstream

[Looks like upstream no longer even depends on time.](https://github.com/Builditluc/wiki-tui/blob/main/Cargo.lock)

File: pkgs/misc/wiki-tui/default.nix (hunk 3 above)
Comment (Builditluc) [reply pkgs/misc/wiki-tui/default.nix:20]: Just out of curiosity, why to we no longer do this?

**Comment by Builditluc** [comment]
@alyssais I've tried updating the bare minimum, but that leads to several version conflicts, so it seems this is the only way

File: pkgs/misc/wiki-tui/default.nix (hunk 3 above)
Comment (alyssais) [reply pkgs/misc/wiki-tui/default.nix:20]: https://github.com/NixOS/nixpkgs/issues/277994

**Comment by Builditluc** [comment]
btw, should the package be moved to `pkgs/by-name`?

**Comment by alyssais** [comment]
> btw, should the package be moved to `pkgs/by-name`?

Up to you, but the more changes are in this PR, the less easy it is to review it quickly, and the motivation here is to prevent an imminent breakage…

**Comment by Builditluc** [comment]
@alyssais yep makes sense, thanks. I think I'll leave it with that and move it in the next version PR

**Pull request merged**